sympy/sets/sets.py

```diff
@@ -210,7 +210,8 @@ def symmetric_difference(self, other):
          return SymmetricDifference(self, other)
 
     def _symmetric_difference(self, other):
-        return Union(Complement(self, other), Complement(other, self))
+        if isinstance(other, FiniteSet) or isinstance(other, Interval):
```

Comment:
Why this change? Complement and Union can work for many other different classes of sets.